src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(
+    name,
+    health = 100,
+  ) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(
+    health,
+    name,
+    hidden = false,
+  ) {
+    super(health, name);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (herbivore instanceof Herbivore && !herbivore.hidden) {
+      herbivore.health -= 50;
+    }
+
+    Animal.alive = Animal.alive.filter(animal => {
+      if (animal instanceof Herbivore && animal.health === 0) {
+        return false;
+      }
+
+      return true;
+    });
```

Comment:
do some check, is health < or = then 0.
And there is no need to make this condition `animal instanceof Herbivore && animal.health === 0`